### PR TITLE
[Linkedin] OAuth 2.0 implementation

### PR DESCRIPTION
**Pull request by kbighorse**
using OAuth 2.0

### CODE REVIEW

**Comment by kbighorse** [comment]
Not fully working yet; in step 3, currently redirects to ```https://www.example.com/endpoint?code=XXX" and displays a HybridAuth information page.

**Comment by kbighorse** [comment]
Running https://github.com/hybridauth/hybridauth/blob/master/hybridauth/Hybrid/Endpoint.php#L59, when should run either https://github.com/hybridauth/hybridauth/blob/master/hybridauth/Hybrid/Endpoint.php#L51 or https://github.com/hybridauth/hybridauth/blob/master/hybridauth/Hybrid/Endpoint.php#L55.

For some reason, request is missing ```hauth_start``` or ```hauth_done``` params, presumably ```hauth_done``` if this is Step 3.

File: hybridauth/Hybrid/Providers/LinkedIn.php [hunk 1]
```diff
@@ -7,289 +7,45 @@
...
+            $this->api->access_token            = $this->token("access_token");
+            $this->api->refresh_token           = $this->token("refresh_token");
+            $this->api->access_token_expires_in = $this->token("expires_in");
+            $this->api->access_token_expires_at = $this->token("expires_at");
+        }
 
+        // Provider api end-points
+        $this->api->authorize_url  = "https://www.linkedin.com/oauth/v2/authorization";
+        $this->api->token_url      = "https://www.linkedin.com/oauth/v2/accessToken";
+        $this->api->token_info_url = "https://www.linkedin.com/oauth/v2";
```
Comment (kbighorse) [inline hybridauth/Hybrid/Providers/LinkedIn.php]: not sure where this is used.

File: hybridauth/Hybrid/Providers/LinkedIn.php [hunk 2]
```diff
@@ -48,4 +48,23 @@ function initialize() {
         $this->api->token_url      = "https://www.linkedin.com/oauth/v2/accessToken";
         $this->api->token_info_url = "https://www.linkedin.com/oauth/v2";
     }
+
+    /**
+     * load the user profile from the IDp api client
+     */
+    function getUserProfile() {
+        $this->api->curl_header = array(
```
Comment (kbighorse) [inline hybridauth/Hybrid/Providers/LinkedIn.php]: perhaps there is a better place to do this for other requests? @dashohoxha

Comment (dashohoxha) [inline hybridauth/Hybrid/Providers/LinkedIn.php]: If it works, it is OK. This is the best place in my opinion.

**Comment by StorytellerCZ** [comment]
@kbighorse How is it going?

**Comment by ApacheEx** [comment]
I'd like to help, so I've already migrated to OAuth 2.0. Check it out #803.

**Comment by kbighorse** [comment]
Sorry I missed you, @StorytellerCZ, I was done. But looks like things worked out with @ApacheEx's PR.

**Pull request closed without merging**